### ISSUE TITLE
Extend the TupleGenerator for generating Tuple-dependent code in other classes

### ISSUE THREAD

**Issue by twalthr** [stratosphere/stratosphere#597]
Some classes also depend on the number of tuples. 
E.g. the `types()` of `CsvReader` or `CLASSES` of `TupleTypeInfo`

@StephanEwen already suggested to extend the TupleGenerator. I would like to do this but I'm not sure how.

What do you think about adding a special comment to these classes so that the tuple generator knows where to put Tuple-dependent code, instead of generating the whole class?

```
class CsvReader {
    // Constructor, methods etc.
    // ......

    // BEGIN_OF_TUPLE_DEPENDENT_CODE

    // END_OF_TUPLE_DEPENDENT_CODE
}
```

**Comment by alexff91** [comment]
Hi, maybe better to do this:
Check if there the same methods, if no add tuple-methods to end of class.
This will help if someone will forgot to add // BEGIN_OF_TUPLE_DEPENDENT_CODE at first to the class.

**Comment by fhueske** [comment]
Fixed in b6edba26c3443ebe33a3de4f482900fa424430f2
